dotnet/src/SemanticKernel.Core/Search/TextSearchExtensions.cs

```diff
@@ -0,0 +1,285 @@
+            };
+
+            var result = await textSearch.GetTextSearchResultsAsync(query?.ToString()!, searchOptions, cancellationToken).ConfigureAwait(false);
+            return await result.Results.ToListAsync(cancellationToken).ConfigureAwait(false);
+        }
+
+        options ??= DefaultGetTextSearchResultsMethodOptions();
+        return KernelFunctionFactory.CreateFromMethod(
+                GetTextSearchResultAsync,
+                options);
+    }
+
+    /// <summary>
+    /// Create a <see cref="KernelFunction"/> which invokes <see cref="ITextSearch.GetSearchResultsAsync(string, TextSearchOptions?, CancellationToken)"/>.
+    /// </summary>
+    /// <param name="textSearch">The ITextSearch instance to use.</param>
+    /// <param name="options">Optional KernelFunctionFromMethodOptions which allow the KernelFunction metadata to be specified.</param>
+    /// <param name="searchOptions">Optional TextSearchOptions which override the options provided when the function is invoked.</param>
+    public static KernelFunction CreateGetSearchResults(this ITextSearch textSearch, KernelFunctionFromMethodOptions? options = null, TextSearchOptions? searchOptions = null)
+    {
+        async Task<IEnumerable<object>> GetSearchResultAsync(Kernel kernel, KernelFunction function, KernelArguments arguments, CancellationToken cancellationToken)
+        {
+            arguments.TryGetValue("query", out var query);
+            if (string.IsNullOrEmpty(query?.ToString()))
+            {
+                return [];
+            }
+
+            var parameters = function.Metadata.Parameters;
+
+            searchOptions ??= new()
+            {
+                Count = GetArgumentValue(arguments, parameters, "count", 2),
+                Offset = GetArgumentValue(arguments, parameters, "skip", 0)
+            };
+
+            var result = await textSearch.GetSearchResultsAsync(query?.ToString()!, searchOptions, cancellationToken).ConfigureAwait(false);
+            return await result.Results.ToListAsync(cancellationToken).ConfigureAwait(false);
+        }
+
+        options ??= DefaultGetSearchResultsMethodOptions();
+        return KernelFunctionFactory.CreateFromMethod(
+                GetSearchResultAsync,
+                options);
+    }
+    #endregion
+
+    #region private
+    /// <summary>
+    /// Get the argument value from <see cref="KernelArguments"/> or users default value from
+    /// <see cref="KernelReturnParameterMetadata"/> or default to the provided value.
+    /// </summary>
+    /// <param name="arguments">KernelArguments instance.</param>
+    /// <param name="parameters">List of KernelReturnParameterMetadata.</param>
+    /// <param name="name">Name of the argument.</param>
+    /// <param name="defaultValue">Default value of the argument.</param>
+    private static int GetArgumentValue(KernelArguments arguments, IReadOnlyList<KernelParameterMetadata> parameters, string name, int defaultValue)
```

Comment:
nit: I would leverage the `TryGetValue` result, use pattern matching, and restructure the code slightly to simplify it:
```csharp
private static int GetArgumentValue(KernelArguments arguments, IReadOnlyList<KernelParameterMetadata> parameters, string name, int defaultValue)
{
    if (arguments.TryGetValue(name, out var value) && value is int argument)
    {
        return argument;
    }

    value = parameters.FirstOrDefault(parameter => parameter.Name == name)?.DefaultValue;
    if (value is int metadataDefault)
    {
        return metadataDefault;
    }

    return defaultValue;
}
```